ruleset/rules/0690-gcp_rules.xml

```diff
@@ -339,4 +578,35 @@ ID: 65000 - 65499
         <description>GCP emergency event on project $(gcp.resource.labels.project_id), monitored resource type: $(gcp.resource.type)</description>
         <options>no_full_log</options>
     </rule>
+
+    <!-- VPC Firewall rules -->
+    <rule id="65047" level="3">
+        <if_sid>65000</if_sid>
+        <field name="gcp.jsonPayload.rule_details.reference" type="pcre2">^(?!^$).*</field>
+        <description>GCP VPC Firewall event</description>
+        <options>no_full_log</options>
+    </rule>
+
+    <rule id="65048" level="5">
+        <if_sid>65047</if_sid>
+        <field name="gcp.jsonPayload.rule_details.action">DENY</field>
+        <description>GCP VPC Firewall: DENY rule triggered</description>
+        <options>no_full_log</options>
```

Comment:
I think Mitre T1562.006 can be added here

Reply:
This rules detects a firewall rule triggered with a deny output, T1562.006 would be if the firewall rule has been deleted.